molecularnodes/entities/molecule/base.py

```diff
@@ -60,15 +49,108 @@ def __init__(self, file_path: Union[str, Path, io.BytesIO]):
         file_path : Union[str, Path, io.BytesIO]
             The file path to the file which stores the atomic coordinates.
         """
-        super().__init__()
-        self._parse_filepath(file_path=file_path)
-        self.file: str
-        self.array: np.ndarray
-        self._frames_collection: str | None
+        self._frames_collection: str | None = None
+        self._code: str | None = None
         self._entity_type = EntityType.MOLECULE
+        super().__init__()
+        self._read(file_path)
+        self.atom_array: AtomArray | AtomArrayStack
+        self._create_object()
+
+    @property
+    def code(self) -> str | None:
+        """
+        Get the code for the molecule.
+        """
+        return self._code
+
+    def _read(self, file_path: str | Path | io.BytesIO) -> None:
```

Comment:
Question: why not just load in the data? E.g. instead of returning a handle,  why not just create the AtomArray with the fields of interest. You could optionally set the handle as well. Seems like an extra bit of indirection